tests/perf/prerequisites/istio/setup.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -e
+
+WORKING_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+if [[ -z "${NAMESPACE}" ]]; then
+  export NAMESPACE=istio-perf-test
+fi
+
+CLUSTER_DOMAIN_NAME=$(kubectl get gateways.networking.istio.io kyma-gateway -n kyma-system -ojsonpath="{.spec.servers[0].hosts[0]}" | sed 's/*.//g' )
+export CLUSTER_DOMAIN_NAME
+
+export WORKLOAD_SIZE=${WORKLOAD_SIZE:-10}
+export VUS=${VUS:-64}
+
+common_resources=(
+    namespace.yaml
+)
+
+workload_resources=(
+    app.yaml
+    api.yaml
+)
+
+for resource in "${common_resources[@]}"; do
+    envsubst <"${WORKING_DIR}/$resource" | kubectl -n "${NAMESPACE}" apply -f -
+done
+
+for (( i = 0; i < WORKLOAD_SIZE; i++ )); do
+    export WORKER=$((i + 1))
+    for resource in "${workload_resources[@]}"; do
```

Comment:
Only one of the files actually starts k8s pods: `app.yaml`. The other, `apirule.yaml` only applies apiRules. But we're waiting for pods in both cases. I'd suggest to extract the entire loop into a function. This function could have two params: `resourceName:string` and `waitForPods: bool`. Then we can invoke this function twice: `doIt("app.yaml", "true")` and doIt("apirule.yaml", "false")`. Perhaps you can find a better solution.